infrastructure/httpapi/handlers/crossfire.go

```diff
@@ -1,9 +1,162 @@
+
+				validatorStats, err := handler.getValidatorStats(participant.OperatorAddress)
+				if err != nil {
+					continue
+				}
+				validatorDetail := CrossfireValidatorDetails{
+					operatorAddress:            participant.OperatorAddress,
+					initialDelegatorAddress:    dbValidator.InitialDelegatorAddress,
+					moniker:                    dbValidator.Moniker,
+					taskSetup:                  dbValidator.TaskPhase1NodeSetup,
+					taskKeepActive:             dbValidator.TaskPhase2KeepNodeActive,
+					taskProposalVote:           dbValidator.TaskPhase2ProposalVote,
+					taskNetworkUpgrade:         dbValidator.TaskPhase2NetworkUpgrade,
+					rankPhase1n2CommitmentRank: dbValidator.RankTaskPhase1n2CommitmentCount,
+					rankPhase3CommitmentRank:   dbValidator.RankTaskPhase3CommitmentCount,
+					rankTxSentRank:             dbValidator.RankTaskHighestTxSent,
+					stats:                      validatorStats,
+				}
+				finalList = append(finalList, validatorDetail)
+			}
+		}
+	}
+
+	httpapi.Success(ctx, finalList)
+}
+
+func (handler *Crossfire) getValidatorStats(operatorAddress string) (*ValidatorStats, error) {
+	// Fetch Validator Stats
+	key := fmt.Sprintf("%s%s", "%", operatorAddress)
+	validatorStatRows, err := handler.crossfireValidatorsStatsView.FindAllLike(key)
+	if err != nil {
+		return nil, fmt.Errorf("[error-crossfire] Fetching Validator Stats, %v", err)
+	}
+
+	var finalValidatorStats *ValidatorStats
+	for _, validatorStatRow := range validatorStatRows {
+		switch strings.Split(validatorStatRow.Key, crossfire_constants.DB_KEY_SEPARATOR)[0] {
+		case crossfire_constants.TOTAL_TX_SENT_PREFIX:
+			*finalValidatorStats.totalTxSent = validatorStatRow.Value
+		case crossfire_constants.PHASE_1_TX_SENT_PREFIX:
+			*finalValidatorStats.txSentPhase1 = validatorStatRow.Value
+		case crossfire_constants.PHASE_2_TX_SENT_PREFIX:
+			*finalValidatorStats.txSentPhase2 = validatorStatRow.Value
+		case crossfire_constants.PHASE_3_TX_SENT_PREFIX:
+			*finalValidatorStats.txSentPhase3 = validatorStatRow.Value
+		case crossfire_constants.PHASE3_COMMIT:
+			*finalValidatorStats.commitCountPhase3 = validatorStatRow.Value
+		case crossfire_constants.PHASE2_COMMIT:
+			*finalValidatorStats.commitCountPhase2 = validatorStatRow.Value
+		case crossfire_constants.PHASE1_COMMIT:
+			*finalValidatorStats.commitCountPhase1 = validatorStatRow.Value
+		default:
+			break
+		}
+	}
+	return finalValidatorStats, nil
+}
+
+// ValidatorStats Validator statistics
+type ValidatorStats struct {
+	totalTxSent       *int64
+	txSentPhase1      *int64
+	txSentPhase2      *int64
+	txSentPhase3      *int64
+	commitCountPhase1 *int64
+	commitCountPhase2 *int64
+	commitCountPhase3 *int64
+}
+
+// CrossfireValidatorDetails response object
+type CrossfireValidatorDetails struct {
+	operatorAddress            string
+	initialDelegatorAddress    string
+	moniker                    string
```

Comment:
@cdc-Hitesh could you also add following fields for return:

```
- join height
- join block time
- tx sent count
- phase1n2 commitment count
- phase3 commitment count
```

Thanks!

Reply:
Cool noted

Reply:
@foreseaz  Oh It's already there under `stats`  of type `ValidatorStats`. Do you mean something else?

Reply:
Oh sorry, haven't seen that one. Should be fine! Thanks

Reply:
Cool.